File: Safari-609.1.11.1/LayoutTests/http/tests/loading/resources/resource-that-goes-back-while-still-loading.php
<?php

@apache_setenv('no-gzip', 1);
@ini_set('zlib.output_compression', 0);
@ini_set('implicit_flush', 1);
for ($i = 0; $i < ob_get_level(); $i++) { ob_end_flush(); }
ob_implicit_flush(1);

header("HTTP/1.1 200 OK"); flush();

echo str_repeat(" ", 1024)."<pre>"; flush();
echo "This page should trigger a back navigation while it's still loading.<br>\r\n";
echo "<script>setTimeout('window.history.back()', 0);</script>";

while(true)
{
    echo "Still loading...<br>\r\n"; flush(); sleep(1);
}

?>

File: LayoutTests/http/tests/loading/resources/resource-that-goes-back-while-still-loading.py
#!/usr/bin/env python3

import sys
import time

sys.stdout.write(
    'status: 200\r\n'
    'Content-Type: text/html\r\n\r\n'
)

sys.stdout.flush()

sys.stdout.write(' ' * 1024)
sys.stdout.flush()

sys.stdout.write(
    'This page should trigger a back navigation while it\'s still loading.<br>\r\n'
    '<script>setTimeout(\'window.history.back()\', 0);</script>'
)

while True:
    sys.stdout.write('Still loading...<br>\r\n')
    sys.stdout.flush()
    time.sleep(1)